=== modified file 'portato/gui/gui_helper.py'
--- a/portato/gui/gui_helper.py	2007-10-22 22:41:44 +0000
+++ b/portato/gui/gui_helper.py	2008-03-10 17:06:04 +0000
@@ -21,7 +21,7 @@
 # some backend things
 from .. import backend, plugin
 from ..backend import flags, system, set_system
-from ..helper import debug, info, send_signal_to_group, set_log_level, unique_array
+from ..helper import debug, info, error, send_signal_to_group, set_log_level, unique_array
 from ..waiting_queue import WaitingQueue
 
 # parser
@@ -504,8 +504,12 @@
 			self.process = Popen(command+options+packages, stdout = self.pty[1], stderr = STDOUT, shell = False, env = system.get_environment(), preexec_fn = os.setsid)
 			
 			# remove packages from queue
-			for i in it:
-				self.remove_with_children(i)
+			if self.tree.is_in_emerge(it[0]):
+				self.remove_with_children(self.emergeIt)
+			elif self.tree.is_in_unmerge(it[0]):
+				self.remove_with_children(self.unmergeIt)
+			else:
+				error("Iterator neither in emerge nor in unmerge.")
 			
 			# update title
 			old_title = self.console.get_window_title()

